internal/provisioner/model/model.go

```diff
@@ -171,6 +194,19 @@ type ContourSpec struct {
 	// defaults to 0.
 	KubernetesLogLevel uint8
 
+	// An update strategy to replace existing Envoy DaemonSet pods with new pods.
+	// when envoy be running as a `Deployment`,it's must be nil
+	// +optional
+	EnvoyUpdateStrategy appsv1.DaemonSetUpdateStrategy
```

Comment:
It's a little long, but helps with clarity on different between the two similar fields:
```suggestion
	EnvoyDaemonSetUpdateStrategy appsv1.DaemonSetUpdateStrategy
```